pkg/controller/windowsmachine/metrics/metrics.go

```diff
@@ -84,6 +85,20 @@ func Add(ctx context.Context, cfg *rest.Config, namespace string) error {
 		return errors.Wrap(err, "could not create metrics Service")
 	}
 
+	// Create a monitoring client to hold the ServiceMonitor object
+	mclient, err := monclient.NewForConfig(cfg)
+	if err != nil {
+		return errors.Wrap(err, "could not create monitoring client")
+	}
+
+	// In the case of an operator restart, a previous SM object will be deleted and a
+	// new one will be created with the current spec. If a SM object does not yet exist,
+	// one will be created.
+	deleteErr := mclient.ServiceMonitors(namespace).Delete(context.TODO(), windowsMetricsEndpoints, metav1.DeleteOptions{})
+	if deleteErr != nil && !apierrors.IsNotFound(deleteErr) {
+		return errors.Wrap(err, "could not delete existing ServiceMonitor object")
```

Comment:
instead of using `deleteErr` just use `err`
you are returning wrapped `err` anyway :wink:

Reply:
Only reason to use another error variable besides `err` is if you want to preserve a previous error for some reason.